runtime/lua/vim/lsp/semantic_tokens.lua

```diff
@@ -0,0 +1,95 @@
+local M = {}
+
+local semantic_tokens = {}
+
+local function modifiers_from_number(x, modifiers_table)
+  local function get_bit(n, k)
+    -- Based on/from https://stackoverflow.com/a/26230537
+    -- (n >> k) & 1
+    return bit.band(bit.rshift(n, k), 1)
+  end
+
+  local modifiers = {}
+  for i = 0, #modifiers_table - 1 do
+    local bit = get_bit(x, i)
+    if bit == 1 then
+      table.insert(modifiers, 1, modifiers_table[i + 1])
+    end
+  end
+
+  return modifiers
+end
+
+function M._handle_semantic_tokens_full(client_id, bufnr, response)
```

Comment:
since the module is already called semantic tokes I guess you can drop it to make the function names more terse

Reply:
@theHamsta So like `_handle_full()`? Or what did you have in mind exactly?

Reply:
I would prefer it. Especially for `'lsp.semantic_tokens'.get_semantic_tokens()` a `'lsp.semantic_tokens'.get()` would feel better. But I `get` is feels also a bit unsafe for external users since they must know when to request tokens. Maybe a subscription to token changes would be better?

I also played a bit around with this PR: https://github.com/theHamsta/neovim/blob/81f20e363c95437b12eec600eb003a9b54b5d7d4/runtime/lua/vim/lsp/semantic_tokens.lua#L1